src/service.ts

```diff
@@ -1,116 +1,65 @@
-        chain(({term, limit, searchAfter}) => getTransactionBySnapshot(es)(term, limit, searchAfter)),
+        chain(validateAddressesEvent),
+        map(extractTerm),
+        chain(({ termName, termValue }) => getSnapshot(os)(termName, termValue)),
         fold(
             reason => task.of(errorResponse(reason)),
             value => task.of(successResponse(StatusCodes.OK)(value))
         )
     )
 
-export const getTransactionsByAddressHandler = (event: APIGatewayEvent, es: Client) =>
+export const getGlobalSnapshotRewards = (event: APIGatewayEvent, os: Client) =>
     pipe(
         taskEither.of<ApplicationError, APIGatewayEvent>(event),
         chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionByAddress(es)(term, null, limit, searchAfter)),
+        map(extractTerm),
+        chain(({ termName, termValue }) => getSnapshotRewards(os)(termName, termValue)),
         fold(
             reason => task.of(errorResponse(reason)),
             value => task.of(successResponse(StatusCodes.OK)(value))
         )
     )
 
-export const getTransactionsBySenderHandler = (event: APIGatewayEvent, es: Client) =>
+export const getBalanceByAddressHandler = (event: APIGatewayEvent, os: Client) =>
     pipe(
         taskEither.of<ApplicationError, APIGatewayEvent>(event),
         chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionBySender(es)(term, limit, searchAfter)),
+        map(extractOrdinal),
+        chain(({ term, ordinal }) => getBalanceByAddress(os)(term, ordinal)),
         fold(
             reason => task.of(errorResponse(reason)),
             value => task.of(successResponse(StatusCodes.OK)(value))
         )
     )
 
-export const getTransactionsByReceiverHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionByReceiver(es)(term, limit, searchAfter)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
\ No newline at end of file
+const extractOrdinal = (event: APIGatewayEvent) => {
+    return {
+        term: event.pathParameters!.term,
+        ordinal: (event.queryStringParameters && event.queryStringParameters!.ordinal) || undefined
+    }
+}
+
+const extractTerm = (event: APIGatewayEvent) => {
+    if (event.pathParameters!.term == 'latest')
+        return {
+            termName: 'ordinal',
+            termValue: 'latest'
+        }
+    return {
+        termName: isNumber(event.pathParameters!.term) ? 'ordinal' : 'hash',
+        termValue: event.pathParameters!.term
+    }
+}
+
+const isNumber = (term: string): boolean => { return /^\d+$/.test(term) }
```

Comment:
There is a built-in function `isNaN` which returns `true` if the argument is not a number. You can reuse that instead of creating your own function.

Reply:
Below is declaration of _sNaN_:
`declare function isNaN(number: number): boolean;`.
Unfortunately it takes number as an argument. I think it is more designed for things like dividing by zero.

Reply:
TS is more strict when it comes to coercion so you need to convert string to number first `isNaN(Number("ABC"))`